source/MaterialXContrib/MaterialXNode/MaterialXData.cpp

```diff
@@ -1,49 +1,102 @@
-#include <MaterialXData.h>
-#include <Util.h>
+#include "MaterialXData.h"
+#include "MaterialXUtil.h"
 
 #include <MaterialXFormat/XmlIo.h>
 #include <MaterialXGenOgsXml/GlslFragmentGenerator.h>
 #include <MaterialXGenShader/Util.h>
 
-MaterialXData::MaterialXData(const std::string& materialXDocumentPath, 
-                             const std::string& elementPath, 
-                             const MaterialX::FileSearchPath& librarySearchPath)
-    : _shaderGenerator(mx::GlslFragmentGenerator::create()),
-      _genContext(_shaderGenerator),
-      _librarySearchPath(librarySearchPath)
+namespace
 {
-    setRenderableElement(materialXDocumentPath, elementPath);
+mx::DocumentPtr createDocument( const std::string& materialXDocumentPath,
+                                const MaterialX::FileSearchPath& librarySearchPath )
+{
+    // Create document
+    mx::DocumentPtr document = mx::createDocument();
+    if (!document)
+    {
+        throw mx::Exception("Failed to create a MaterialX document");
+    }
+
+    // Load libraries
+    static const mx::StringVec libraries = { "stdlib", "pbrlib", "bxdf", "stdlib/genglsl", "pbrlib/genglsl" };
```

Comment:
To be clean the stdlib/genglsl and pbrbli/genglsl libraries should not be loaded here as it depends on the language specified at generation time. "genglsl" is the language.